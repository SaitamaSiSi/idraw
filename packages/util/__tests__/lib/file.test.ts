import {
  downloadImageFromCanvas
} from '../../src/lib/file';


describe('@idraw/util: lib/file', () => {
  
  const canvas = document.createElement('canvas');
  const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
  const opts = {
    width: 600,
    height: 400,
  }
  ctx.clearRect(0, 0, opts.width, opts.height);
  ctx.fillStyle = '#ffffff';
  ctx.fillRect(0, 0, opts.width / 2, opts.height / 2);

  test('downloadImageFromCanvas', async () => { 
    downloadImageFromCanvas(canvas, {
      filename: 'hello',
      type: 'image/png',
    }); 
    expect(canvas).toMatchSnapshot();
  });
  

});

