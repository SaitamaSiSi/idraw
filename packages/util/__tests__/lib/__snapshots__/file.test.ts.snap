// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@idraw/util: lib/file downloadImageFromCanvas 1`] = `<canvas />`;
